platform/default/mbgl/storage/offline_database.cpp

```diff
@@ -136,6 +136,12 @@ mapbox::sqlite::Statement& OfflineDatabase::getStatement(const char* sql) {
     return *it->second;
 }
 
+void OfflineDatabase::batch(BatchedFn&& fn) {
```

Comment:
Instead of exposing a generic function like this, can we pass a list/collection of `Resource`/`Response` pairs and handle iteration over those, and transaction management internally? E.g. have a public `putRegionResources()` function that internally calls `putRegionResource` for every item, wrapping everything in a transaction.